clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -698,8 +698,10 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(int output_
 {
     uint64_t cur_instr = input.reader->get_instruction_ordinal();
     auto &cur_range = input.regions_of_interest[input.cur_region];
-    if (cur_range.stop_instruction != 0 && cur_instr > cur_range.stop_instruction) {
+    if (input.in_cur_region && cur_range.stop_instruction != 0 &&
```

Comment:
So this is the case when we just exited an roi? Comment please.

Reply:
Done.